snap/snapcraft.yaml

```diff
@@ -330,11 +330,18 @@ parts:
 
   mysql:
     plugin: cmake
-    after: [boost]
+    after: [boost, patches]
 
     # Get from https://dev.mysql.com/downloads/mysql/
     source: https://dev.mysql.com/get/Downloads/MySQL-8.0/mysql-boost-8.0.34.tar.gz
     source-checksum: md5/c8cfab52fbde1cca55accb3113c235eb
+    override-pull: |
+      snapcraftctl pull
+      patch -p1 $SNAPCRAFT_PART_SRC/mysys/CMakeLists.txt < $SNAPCRAFT_STAGE/mysql-mysys_CMakeLists.txt.patch
+      patch -p1 $SNAPCRAFT_PART_SRC/sql/memory/aligned_atomic.h < $SNAPCRAFT_STAGE/mysql-sql_memory_aligned_atomic.h.patch
+      patch -p1 $SNAPCRAFT_PART_SRC/unittest/gunit/memory/aligned_atomic-t.cc < $SNAPCRAFT_STAGE/mysql-unittest_gunit_memory_aligned_atomic-t.cc.patch
+      rm $SNAPCRAFT_PART_SRC/include/my_aligned_malloc.h
+      rm $SNAPCRAFT_PART_SRC/mysys/my_aligned_malloc.cc
```

Comment:
You mention that this is a revert of a [single commit in upstream mysql](https://github.com/mysql/mysql-server/commit/be8348a7c3e8510b998a063065b626a459631b32), but here we're making a number of changes manually. Is there a reason we can't use a single patch file for this?

Reply:
Yes there is: I found out that the files on the `mysql-server` repo at `mysql-8.0.34` tag differs a little bit from the files at mysql-boost-8.0.34, so I had to "adapt" the patch, but it worked OK.

Reply:
Ah, okay. I think ideally this would be a single patch with some details in it, like a commit message. Let me see if I can help.

Reply:
Alright, see latest commit

Reply:
Much nicer! How did you manage to do it in the end?

Reply:
Good deal. I just cloned mysql, did the revert in git, and then used `git format-patch` to get the .patch file.

Reply:
And you didn't have to adapt anything for it to apply neatly on the decompressed `mysql-boost-8.0.34.tar.gz`?
I must have missed something then...